### ISSUE TITLE
[Feature Request]: Chat GPT Integration 

### ISSUE THREAD

**Issue by Sibam-Paul** [kartikayasijaa/talk-trove#18]
### Feature Description
.
![image](https://github.com/user-attachments/assets/69c21fa7-79e2-4b30-8f02-d4814210d45a)

Just like in slack
@kartikayasija Can u please explain here

**Comment by kartikayasijaa** [comment]
Explanation:
- We need to integrate Chat GPT. 
- When we type `/` , there should be a dropdown for all our integrations. [only chat gpt for now]
- When user clicks on that, a new modal should come up. User should be able to give prompts to that GPT instance.

**Comment by geetanjalidubey9** [comment]
can you pls assign this task to me

**Comment by Sibam-Paul** [comment]
@geetanjalidubey9 Go for it

**Comment by kartikayasijaa** [comment]
@geetanjalidubey9 Have you started this issue?

**Comment by geetanjalidubey9** [comment]
Yess

**Comment by ChampGupta** [comment]
I would like to contribute to this issue and I'll integrate ChatGPT to this issue. Please consider my request and assign if possible.
Regards

**Comment by geetanjalidubey9** [comment]
yeah you can work on this project because i am working on other issue of same project

**Comment by Sibam-Paul** [comment]
@geetanjalidubey9 do u don't want to work or wanna hand over to @ChampGupta

**Comment by Sibam-Paul** [comment]
@ChampGupta please proceed and if u had any problem please raise Q/A section in discussion

**Comment by Sibam-Paul** [comment]
If u have any problem please open a question/answer tab in the discussion section of this repo plz it would be helpful 
and always look at it for announcement plz of any changes we will make regarding issues

**Comment by ChampGupta** [comment]
@Sibam-Paul I'll complete it ASAP

**Comment by Sibam-Paul** [comment]
@ChampGupta Please see discussion tab and join discord plz

**Comment by vaibhav-yerkar** [comment]
I would like to work on this

**Comment by yashksaini-coder** [comment]
I would like to work on it @Sibam-Paul

**Comment by Sibam-Paul** [comment]
@geetanjalidubey9 inactive assigning to next person

**Comment by Sibam-Paul** [comment]
@vaibhav-yerkar If u want to work tell me

**Comment by vaibhav-yerkar** [comment]
Thanks

**Comment by vaibhav-yerkar** [comment]
Update:
- implemented backend route for the api request
- frontend UI implemented

Issue Facing:
- No free api request available  for chatgpt api , so haven't tested its backend , so i built another route for gemini - backend tested
- integration of frontend with backend api

**Comment by Sibam-Paul** [comment]
@vaibhav-yerkar yes implement it and attact a working 📹

**Comment by vaibhav-yerkar** [comment]
@Sibam-Paul 
working Prototype:
https://github.com/user-attachments/assets/3694db1d-2aed-4410-ac27-7f6bc3bfb433

**Comment by vaibhav-yerkar** [comment]
@Sibam-Paul should I raise a PR or there's any modification to be made

**Comment by Sibam-Paul** [comment]
Where can't find it link it here

**Comment by vaibhav-yerkar** [comment]
@Sibam-Paul I haven't pushed the changes yet, I was asking if I should or there's anything else to be done/implemented before that

**Comment by Sibam-Paul** [comment]
@vaibhav-yerkar bro can U come to vc of discord now

**Comment by Sibam-Paul** [comment]
Just stay in the general of gssoc channel  I will come

**Comment by vaibhav-yerkar** [comment]
@Sibam-Paul can we have meet in evening, maybe around 4

**Comment by Sibam-Paul** [comment]
First join discord see discussion tab

**Comment by Sibam-Paul** [comment]
Around 9 it would be possible join discord my username specter102
